CMakeLists.txt

```diff
@@ -199,19 +199,48 @@ command_line_arg(cppad_tape_addr_type "unsigned int" STRING
 )
 #
 # cppad_debug_which
-command_line_arg(cppad_debug_which "debug_all" STRING
-    "debug_even, debug_odd, debug_all, or debug_none"
+IF( NOT debug_which )
+    SET(debug_which "" CACHE STRING
+        "debug_even, debug_odd, debug_all, or debug_none"
+    )
+ENDIF( NOT debug_which )
+IF( CMAKE_BUILD_TYPE )
+    IF( NOT ${cppad_debug_which} STREQUAL "" )
+        print_variable(CMAKE_BUILD_TYPE)
+        print_variable(cppad_debug_which)
+        MESSAGE(FATAL_ERROR
+            "Both CMAKE_BUILD_TYPE and cppad_debug_which specified"
+        )
+    ENDIF( NOT ${cppad_debug_which} STREQUAL "" )
+    #
+    STRING( TOLOWER "${CMAKE_BUILD_TYPE}" build_type )
+    IF( build_type STREQUAL "debug" )
+        SET(cppad_debug_which "debug_all")
+    ELSEIF( build_type STREQUAL "release" )
+        SET(cppad_debug_which "debug_none")
+    ELSE( build_type )
+        print_variable(CMAKE_BUILD_TYPE)
+        MESSAGE(FATAL_ERROR "CMAKE_BUILD_TYPE is not debug or release")
```

Comment:
Why do you output a `FATAL_ERROR`. Would you like to also consider `RelWithDebInfo` for instance?

Reply:
> Why do you output a `FATAL_ERROR`. Would you like to also consider `RelWithDebInfo` for instance?

Good point !
But this is a more complicated case becasue RelWithDebug info does not map simply to the current cppad_debug_which flags (which is what actually gets used when deciding how to compile each individual file). 
We could add another cppad_debug_which value that corresponded to this.

The important placed for this is when building cppad_lib which gets installed. In the file
https://github.com/coin-or/CppAD/blob/master/cppad_lib/CMakeLists.txt
see the call to
set_compile_flags(cppad_lib "${cppad_debug_which}" "${source_list}" )
and the macro
https://github.com/coin-or/CppAD/blob/master/cmake/set_compile_flags.cmake
where the real work is done.

Reply:
I would just remove the FATAL_ERROR and mention that we only consider the `debug_none` case. 
What do you think?